### ISSUE TITLE
[YSQL] Tserver Error: WaitForYsqlBackendsCatalogVersion Failed - Unexpected Number of Lagging Backends

### ISSUE THREAD

**Issue by shishir2001-yb** [yugabyte/yugabyte-db#20965]
Jira Link: [DB-9940](https://yugabyte.atlassian.net/browse/DB-9940)
### Description

Tried on version: **2.21.1.0-b3**

During the execution of the Cross DB DDL sample app encountered the following Tserver error
`
E0206 08:09:12.704667 39551 tablet_service.cc:2021] P 177259429e4b42e5b93c32a2838579a3: WaitForYsqlBackendsCatalogVersion: num lagging backends query failed: Internal error (yb/tserver/tablet_service.cc:2005): Unexpected prev_num_lagging_backends: 1, num_lagging_backends: 2
`

**Test details**:
`
Start the Cross-DB Concurrent DDLs sample app with 20 write threads, 5 read threads and 20 databases(10 colocated and 10 non-colocated) which will execute DDLs in parallel with DMLs across different databases. Notably, we ensure there is no simultaneous execution of DDL operations within the same database.
`

List of DDLs:
```
private static List<List<String>> ddlList = List.of(
            List.of("CREATE INDEX idx1 ON ? (k)", "DROP INDEX idx1"),
            List.of("CREATE TABLE tempTable1 AS SELECT * FROM ? limit 1000000", "ALTER TABLE tempTable1 RENAME TO tempTable1_new", "DROP TABLE tempTable1_new"),
            List.of("CREATE MATERIALIZED VIEW mv1 as SELECT k from ? limit 10000", "REFRESH MATERIALIZED VIEW mv1", "DROP MATERIALIZED VIEW mv1"),
            List.of("ALTER TABLE ? ADD newColumn1 TEXT DEFAULT 'dummyString'", "ALTER TABLE ? DROP newColumn1"),
            List.of("ALTER TABLE ? ADD newColumn2 TEXT NULL", "ALTER TABLE ? DROP newColumn2"),
            List.of("CREATE VIEW view1_? AS SELECT k from ?", "DROP VIEW view1_?"),
            List.of("ALTER TABLE ? ADD newColumn3 TEXT DEFAULT 'dummyString'", "ALTER TABLE ? ALTER newColumn3 TYPE VARCHAR(1000)", "ALTER TABLE ? DROP newColumn3"),
            List.of("CREATE TABLE tempTable2 AS SELECT * FROM ? limit 1000000", "CREATE INDEX idx2 ON tempTable2(k)", "ALTER TABLE ? ADD newColumn4 TEXT DEFAULT 'dummyString'", "ALTER TABLE tempTable2 ADD newColumn2 TEXT DEFAULT 'dummyString'", "TRUNCATE table ? cascade", "ALTER TABLE ? DROP newColumn4", "ALTER TABLE tempTable2 DROP newColumn2", "DROP INDEX idx2", "DROP TABLE tempTable2"),
            List.of("CREATE VIEW view2_? AS SELECT k from ?", "CREATE MATERIALIZED VIEW mv2 as SELECT k from ? limit 10000", "REFRESH MATERIALIZED VIEW mv2", "DROP MATERIALIZED VIEW mv2", "DROP VIEW view2_?")
 );

```

`
Not sure if this could be related somehow, but after 5 minutes, one of the insert query failed with the following error:
ERROR: Resource unavailable: null
`

Logs: [https://drive.google.com/file/d/1zu3B9GPiK_3pAeNvaCCcQyw0bdvjRWdT/view?usp=sharing](https://drive.google.com/file/d/1zu3B9GPiK_3pAeNvaCCcQyw0bdvjRWdT/view?usp=sharing)

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-9940]: https://yugabyte.atlassian.net/browse/DB-9940?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by myang2021** [comment]
per @jaki slack message:

> I'm seeing other DDLs timeout as well
> 
> `Exception occurred during DDL postgres_4 DDL query ALTER TABLE tempTable2_tb_0 ADD newColumn5 TEXT DEFAULT 'dummyString': ERROR: Timed out waiting for Alter Table
> `
> 
> 
> 
> If the system is that much overloaded, perhaps it is expected
> 
> 
> 
Looks like master is overloaded so it is causing various DDLs to time out, not just create index.

**Comment by rthallamko3** [comment]
@shishir2001-yb , Is this duplicate of https://github.com/yugabyte/yugabyte-db/issues/20965?

**Comment by myang2021** [comment]
@rthallamko3 I closed this ticket by mistake and that's why I reopened it, are you referring to another ticket as duplicate of this one?
